concrete/src/Http/Middleware/ThumbnailMiddleware.php

```diff
@@ -59,7 +70,9 @@ public function process(Request $request, DelegateInterface $frame)
     {
         $response = $frame->next($request);
 
-        if ($this->app->isInstalled()) {
+        // if the thumbnail generator is async, we do not use the thumbnail middleware.
+
+        if ($this->app->isInstalled() && $this->config->get('concrete.misc.basic_thumbnailer_generation_strategy') == 'now') {
```

Comment:
No – we only use the thumbnail middleware if we're NOT using the async method, but the async method explicitly runs through its own AJAX route.